runtime/packages/core-modules/assembly/log.ts

```diff
@@ -1,15 +1,43 @@
-import { Serialiser } from '@as-substrate/core-utils';
-import { ScaleString } from 'as-scale-codec';
-import { ext_misc_print_utf8_version_1 } from './env';
+import { Serialiser, Utils } from '@as-substrate/core-utils';
+import { ext_misc_print_utf8_version_1, ext_logging_log_version_1 } from './env';
 
 export class Log {
+    public static readonly LOG_TARGET: string = "runtime";
+    public static readonly ERROR_LEVEL: i32 = 0;
+    public static readonly WARN_LEVEL: i32 = 1;
     /**
-     * Prints a message to the Host
+     * Base function for logging
+     * @param logLevel 
+     * @param message 
+     */
+    static baseLog(logLevel: i32, message: string): void {
+        const messageU8a: u8[] = Utils.stringsToU8a([message]);
+        const logTarget = Serialiser.serialiseResult(Utils.stringsToU8a([Log.LOG_TARGET]));
+        const encodedMessage = Serialiser.serialiseResult(messageU8a);
+        ext_logging_log_version_1(logLevel, logTarget, encodedMessage);
+    }
+
+    /**
+     * Prints a message to the Host (log level is info and debug)
      * @param message message to log
      */
-    static printUtf8(message: string): void{
-        const msg: ScaleString = new ScaleString(message);
-        const encodedMessage = Serialiser.serialiseResult(msg.toU8a().slice(1, msg.encodedLength()));
+    static printUtf8(message: string): void {
```

Comment:
I think that for the sake of consistency you can rename this function to `info`

Reply:
Question: Is there a reason for which we use `ext_logging_log_version_1` instead of `ext_misc_print_utf8_version_1`